scheduler/group/group.go

```diff
@@ -0,0 +1,61 @@
+package group
+
+import (
+	"runtime"
+
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/executor"
+	"polydawn.net/repeatr/lib/guid"
+	"polydawn.net/repeatr/scheduler"
+)
+
+// interface assertion
+var _ scheduler.Scheduler = &Scheduler{}
+
+// Dumb struct to send job references back
+type hold struct {
+	forumla  def.Formula
+	response chan def.Job
+}
+
+type Scheduler struct {
+	groupSize int
+	executor  *executor.Executor
+	queue     chan *hold
+}
+
+func (s *Scheduler) Configure(e *executor.Executor) {
+	s.groupSize = runtime.NumCPU()
+	s.executor = e
+	s.queue = make(chan *hold)
+}
+
+func (s *Scheduler) Start() {
+	for w := 1; w <= s.groupSize; w++ {
+		go s.Run()
+	}
+}
+
+func (s *Scheduler) Schedule(f def.Formula) <-chan def.Job {
+
+	h := &hold{
+		forumla:  f,
+		response: make(chan def.Job),
+	}
+
+	go func() {
+		s.queue <- h
+	}()
```

Comment:
+1, that'll do.  I don't think there's any reason not to set the buffer size in the thousands, either (afaik there's no serious pre-allocations, though maybe double check me).

(Tangentially now: "tens of thousands" can be situational.  Tens of thousands of goroutines floating "somewhere"?  Fairly fine.  Tens of thousands of them blocking on a single unbuffered queue, forcing all of them to be considered during a single wakeup?  Might be hitting paths that force the scheduler to do a lot of rethinks and get `O(painful)` at much lower numbers.)

I think the spec says you can actually `len()` on a chan.  Not saying it's going to be "thread-safe" (making choices on it instead of an attempt-and-handle-errors would be a TOCTOU), but it does offer rough inspection.